principles/1_SecurityImplementation.md

```diff
@@ -0,0 +1,61 @@
+* MSTG-ARCH-4 Data considered sensitive in the context of the mobile app is clearly identified.
+* MSTG-ARCH-5 All app components are defined in terms of the business functions and/or security functions they provide.
+* MSTG-ARCH-6 A threat model for the mobile app and the associated remote services has been produced that identifies potential threats and countermeasures.
+* MSTG-ARCH-8 There is an explicit policy for how cryptographic keys (if any) are managed, and the lifecycle of cryptographic keys is enforced. Ideally, follow a key management standard such as NIST SP 800-57.
+* MSTG-ARCH-10 Security is addressed within all parts of the software development lifecycle.
+* MSTG-CRYPTO-1 The app does not rely on symmetric cryptography with hardcoded keys as a sole method of encryption.
+* MSTG-CRYPTO-2 The app uses proven implementations of cryptographic primitives.
+* MSTG-CRYPTO-3 The app uses cryptographic primitives that are appropriate for the particular use-case, configured with parameters that adhere to industry best practices.
+* MSTG-CRYPTO-4 The app does not use cryptographic protocols or algorithms that are widely considered deprecated for security purposes.
+* MSTG-CRYPTO-5 The app doesn’t re-use the same cryptographic key for multiple purposes.
+* MSTG-CRYPTO-6 All random values are generated using a sufficiently secure random number generator.
+* MSTG-NETWORK-1 Data is encrypted on the network using TLS. The secure channel is used consistently throughout the app.
+* MSTG-NETWORK-2 The TLS settings are in line with current best practices, or as close as possible if the mobile operating system does not support the recommended standards.
+* MSTG-NETWORK-3 The app verifies the X.509 certificate of the remote endpoint when the secure channel is established. Only certificates signed by a trusted CA are accepted.
+* MSTG-NETWORK-6 The app only depends on up-to-date connectivity and security libraries.
+* MSTG-PLATFORM-1 The app only requests the minimum set of permissions necessary
+* MSTG-PLATFORM-2 All inputs from external sources and the user are validated and if necessary sanitized. This includes data received via the UI, IPC mechanisms such as intents, custom URLs, and network sources.
+* MSTG-PLATFORM-3 The app does not export sensitive functionality via custom URL schemes, unless these mechanisms are properly protected.
+* MSTG-PLATFORM-4 The app does not export sensitive functionality through IPC facilities, unless these mechanisms are properly protected.
+* MSTG-PLATFORM-8 Object deserialization, if any, is implemented using safe serialization APIs.
+* MSTG-CODE-1 The app is signed and provisioned with a valid certificate, of which the private key is properly protected.
+* MSTG-CODE-2 The app has been built in release mode, with settings appropriate for a release build (e.g. non-debuggable).
```

Comment:
Consider including MSTG-CODE-3 here after all?
MSTG-CODE-3 Debugging symbols have been removed from native binaries.
We discussed it a little in my doc, and were hesitant it applied, but it's good practice anyway and easy to do. Plus it's a MASVS-level 1 requirement.

Reply:
added